clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -439,25 +454,36 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         // Ensure signal handlers return to the interruption point.
         if (shard->prev_xfer_marker_.marker.marker_type ==
             TRACE_MARKER_TYPE_KERNEL_XFER) {
+            // We use the values popped from the signal-related stacks at the last
+            // TRACE_MARKER_TYPE_KERNEL_XFER marker.
             report_if_false(
                 shard,
-                ((memref.instr.addr == shard->prev_xfer_int_pc_.top() ||
-                  // DR hands us a different address for sysenter than the
-                  // resumption point.
-                  shard->pre_signal_instr_.top().instr.type ==
-                      TRACE_TYPE_INSTR_SYSENTER) &&
+                ((
+                     // Skip this check if we did not see the corresponding
+                     // kernel_event marker in the trace because the trace
+                     // started mid-signal.
+                     shard->last_xfer_int_pc_ == 0 ||
+                     memref.instr.addr == shard->last_xfer_int_pc_ ||
```

Comment:
With the large comments on the other conditions around this, I found it hard to identify this as its own thing rather than part of the condition above.  Maybe put this first, or add a comment saying this is the regular check for post==marker.

Reply:
Added comment.